lib/tapioca/generator.rb

```diff
@@ -550,6 +550,7 @@ def compile_gem_rbi(gem)
         say("Done (empty output)", :yellow)
       else
         content << rbi_body_content
+        content = merge_with_copied_rbi(gem, content)
```

Comment:
While this is fine, I'd rather not do this at this stage. By the time `compiler.compile(gem)` is done, the RBI tree it has built has already been rewritten and transformed. That means, we need to parse it again to do the merge. But, I also suspect that we might be losing some of our transformations we are doing inside `.compile` to serialize the RBI file as a result as well.

In my opinion, it would be cleaner for `compiler.compile(gem, exported_rbi_tree)` be the call and for `SymbolGenerator#generate` to do the actual tree merges, and then apply the RBI transformations only after the merge is done.

I am not sure how we can return/display the merge conflicts in that case, but maybe conflicts can be yielded to a block supplied to the `.compile` method.

So, roughly this `compile_gem_rbi` method might look something like this:
```ruby
def compile_gem_rbi(gem)
  exported_rbi_tree = copy_gem_rbi(gem)
  compiler = Compilers::SymbolTableCompiler.new

  # some other code....

  rbi_body_content = compiler.compile(gem, exported_rbi_tree) do |conflicts|
    # output conflicts as warning here
  end
  # the rest of the method
end
```

What would be even much better, would be if `Gem` actually had a way of returning the exported RBI tree. Something like `gem.exported_rbi_tree` reader that memoized the result. If that existed, then you wouldn't even need to pass an extra argument to `compile`, `SymbolGenerator#generate` could just ask `gem` for it.